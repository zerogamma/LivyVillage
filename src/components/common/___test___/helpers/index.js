import React from 'react'
import renderer from 'react-test-renderer'
import theme from '../../../../utils/style/theme'
import { ThemeProvider } from 'styled-components'

export function renderWithTheme(component) {
    return renderer.create(
        <ThemeProvider theme={theme}>{component}</ThemeProvider>
    )
}
