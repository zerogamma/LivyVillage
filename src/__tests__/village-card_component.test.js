import React from "react"
import VillageCard from "../components/village/village-card";
import VillageList from "../components/village/village-list";


import 'jest-styled-components';
import { renderWithTheme } from '../components/common/___test___/helpers';

describe("village-card Component render test", () => {
    it("renders village-card component correctly", () => {
      const data = {
        "id": 1,
        "name": "Fizkin Voidbuster",
        "thumbnail": "http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg",
        "age": 288,
        "weight": 35.279167,
        "height": 110.43628,
        "hair_color": "Green",
        "professions": [
        "Brewer",
        "Medic",
        "Prospector",
        "Gemcutter",
        "Mason",
        "Tailor"
        ],
        "friends": []
      }
      const filterData = {'age':'', 'weight':'','height':'','hair':'','profession':[]}

      const comp = renderWithTheme(<VillageCard data={data} friendSearch={()=>{}} searchProfession={()=>{}} currentFilter={filterData}/>).toJSON()
      expect(comp).toMatchSnapshot()
    })
})


describe("village-list Component render test", () => {
  it("renders village-list component correctly", () => {
    const data = [{
      "id": 1,
      "name": "Fizkin Voidbuster",
      "thumbnail": "http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg",
      "age": 288,
      "weight": 35.279167,
      "height": 110.43628,
      "hair_color": "Green",
      "professions": [
      "Brewer",
      "Medic",
      "Prospector",
      "Gemcutter",
      "Mason",
      "Tailor"
      ],
      "friends": []
    },
    {
      "id": 2,
      "name": "Malbin Chromerocket",
      "thumbnail": "http://www.publicdomainpictures.net/pictures/30000/nahled/maple-leaves-background.jpg",
      "age": 166,
      "weight": 35.88665,
      "height": 106.14395,
      "hair_color": "Red",
      "professions": [
      "Cook",
      "Baker",
      "Miner"
      ],
      "friends": [
      "Fizwood Voidtossle"
      ]
      },
    ]
    const filterData = {'age':'', 'weight':'','height':'','hair':'','profession':[]}
    
    const comp = renderWithTheme(<VillageList data={data} friendSearch={()=>{}} searchProfession={()=>{}} currentFilter={filterData}/>).toJSON()
    expect(comp).toMatchSnapshot()
  })
})
