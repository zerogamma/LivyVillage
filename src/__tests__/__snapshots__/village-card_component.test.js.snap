// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`village-card Component render test renders village-card component correctly 1`] = `
.c0 {
  display: grid;
  border: 1px solid;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  padding: 5px;
  margin: 15px;
  grid-template-columns: 40% 60%;
  gap: 1px;
  width: 400px;
  height: 550px;
}

.c1 {
  background-size: 100% 100% !important;
}

.c2 span {
  font-variant: all-small-caps;
}

.c3 {
  margin: 10px 0 !important;
  border-bottom: 1px solid;
}

.c3 span {
  font-variant: all-small-caps;
}

.c6 {
  margin-left: 90px;
}

.c4 {
  margin: 0 15px;
}

.c5 {
  border-bottom: 1px solid;
  margin-bottom: 12px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  display: none;
}

@media (max-width:767px) {
  .c0 {
    grid-template-columns: auto;
    grid-template-rows: 25% 75%;
    width: 250px;
    height: 650px;
  }
}

@media (max-width:767px) {
  .c2 {
    font-size: 12px !important;
  }
}

@media (max-width:767px) {
  .c3 {
    font-size: 12px !important;
  }
}

@media (max-width:767px) {
  .c6 {
    margin-left: 120px;
  }
}

<div
  className="MuiPaper-root MuiCard-root c0 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardMedia-root c1"
    style={
      Object {
        "backgroundImage": "url(\\"http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg\\")",
      }
    }
  />
  <div
    className=""
  >
    <div
      className="MuiCardContent-root "
    >
      <h2
        className="MuiTypography-root c2 MuiTypography-h6 MuiTypography-gutterBottom"
      >
        Fizkin Voidbuster
      </h2>
      <p
        className="MuiTypography-root c2 MuiTypography-body1"
      >
        <span>
          Age: 
        </span>
        288
      </p>
      <p
        className="MuiTypography-root c2 MuiTypography-body1"
      >
        <span>
          Weight: 
        </span>
        35.279167
      </p>
      <p
        className="MuiTypography-root c2 MuiTypography-body1"
      >
        <span>
          Height: 
        </span>
        110.43628
      </p>
      <p
        className="MuiTypography-root c2 MuiTypography-body1"
      >
        <span>
          Hair Color: 
        </span>
        Green
      </p>
      <p
        className="MuiTypography-root c3 MuiTypography-body1"
      >
        Friends
      </p>
    </div>
    <div
      className="c4"
    >
      <div
        className="c5"
        onClick={[Function]}
      >
        <p
          className="MuiTypography-root c2 MuiTypography-body1"
        >
          <span>
            Profession
          </span>
        </p>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root c6"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </div>
      <div
        className="c7"
      >
        <p
          className="MuiTypography-root c2 MuiTypography-body1"
        >
          Brewer
        </p>
        <p
          className="MuiTypography-root c2 MuiTypography-body1"
        >
          Medic
        </p>
        <p
          className="MuiTypography-root c2 MuiTypography-body1"
        >
          Prospector
        </p>
        <p
          className="MuiTypography-root c2 MuiTypography-body1"
        >
          Gemcutter
        </p>
        <p
          className="MuiTypography-root c2 MuiTypography-body1"
        >
          Mason
        </p>
        <p
          className="MuiTypography-root c2 MuiTypography-body1"
        >
          Tailor
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`village-list Component render test renders village-list component correctly 1`] = `
Array [
  .c1 {
  display: grid;
  border: 1px solid;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  padding: 5px;
  margin: 15px;
  grid-template-columns: 40% 60%;
  gap: 1px;
  width: 400px;
  height: 550px;
}

.c2 {
  background-size: 100% 100% !important;
}

.c3 span {
  font-variant: all-small-caps;
}

.c4 {
  margin: 10px 0 !important;
  border-bottom: 1px solid;
}

.c4 span {
  font-variant: all-small-caps;
}

.c7 {
  margin-left: 90px;
}

.c5 {
  margin: 0 15px;
}

.c6 {
  border-bottom: 1px solid;
  margin-bottom: 12px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c8 {
  display: none;
}

.c0 {
  -webkit-flex: 0 1 150px;
  -ms-flex: 0 1 150px;
  flex: 0 1 150px;
}

@media (max-width:767px) {
  .c1 {
    grid-template-columns: auto;
    grid-template-rows: 25% 75%;
    width: 250px;
    height: 650px;
  }
}

@media (max-width:767px) {
  .c3 {
    font-size: 12px !important;
  }
}

@media (max-width:767px) {

}

@media (max-width:767px) {
  .c4 {
    font-size: 12px !important;
  }
}

@media (max-width:767px) {
  .c7 {
    margin-left: 120px;
  }
}

<div
    className="c0"
  >
    <div
      className="MuiPaper-root MuiCard-root c1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiCardMedia-root c2"
        style={
          Object {
            "backgroundImage": "url(\\"http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg\\")",
          }
        }
      />
      <div
        className=""
      >
        <div
          className="MuiCardContent-root "
        >
          <h2
            className="MuiTypography-root c3 MuiTypography-h6 MuiTypography-gutterBottom"
          >
            Fizkin Voidbuster
          </h2>
          <p
            className="MuiTypography-root c3 MuiTypography-body1"
          >
            <span>
              Age: 
            </span>
            288
          </p>
          <p
            className="MuiTypography-root c3 MuiTypography-body1"
          >
            <span>
              Weight: 
            </span>
            35.279167
          </p>
          <p
            className="MuiTypography-root c3 MuiTypography-body1"
          >
            <span>
              Height: 
            </span>
            110.43628
          </p>
          <p
            className="MuiTypography-root c3 MuiTypography-body1"
          >
            <span>
              Hair Color: 
            </span>
            Green
          </p>
          <p
            className="MuiTypography-root c4 MuiTypography-body1"
          >
            Friends
          </p>
        </div>
        <div
          className="c5"
        >
          <div
            className="c6"
            onClick={[Function]}
          >
            <p
              className="MuiTypography-root c3 MuiTypography-body1"
            >
              <span>
                Profession
              </span>
            </p>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root c7"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
          <div
            className="c8"
          >
            <p
              className="MuiTypography-root c3 MuiTypography-body1"
            >
              Brewer
            </p>
            <p
              className="MuiTypography-root c3 MuiTypography-body1"
            >
              Medic
            </p>
            <p
              className="MuiTypography-root c3 MuiTypography-body1"
            >
              Prospector
            </p>
            <p
              className="MuiTypography-root c3 MuiTypography-body1"
            >
              Gemcutter
            </p>
            <p
              className="MuiTypography-root c3 MuiTypography-body1"
            >
              Mason
            </p>
            <p
              className="MuiTypography-root c3 MuiTypography-body1"
            >
              Tailor
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  .c1 {
  display: grid;
  border: 1px solid;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  padding: 5px;
  margin: 15px;
  grid-template-columns: 40% 60%;
  gap: 1px;
  width: 400px;
  height: 550px;
}

.c2 {
  background-size: 100% 100% !important;
}

.c3 span {
  font-variant: all-small-caps;
}

.c5 span {
  font-variant: all-small-caps;
}

.c4 {
  margin: 10px 0 !important;
  border-bottom: 1px solid;
}

.c4 span {
  font-variant: all-small-caps;
}

.c8 {
  margin-left: 90px;
}

.c6 {
  margin: 0 15px;
}

.c7 {
  border-bottom: 1px solid;
  margin-bottom: 12px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9 {
  display: none;
}

.c0 {
  -webkit-flex: 0 1 150px;
  -ms-flex: 0 1 150px;
  flex: 0 1 150px;
}

@media (max-width:767px) {
  .c1 {
    grid-template-columns: auto;
    grid-template-rows: 25% 75%;
    width: 250px;
    height: 650px;
  }
}

@media (max-width:767px) {
  .c3 {
    font-size: 12px !important;
  }
}

@media (max-width:767px) {
  .c5 {
    font-size: 12px !important;
  }
}

@media (max-width:767px) {
  .c4 {
    font-size: 12px !important;
  }
}

@media (max-width:767px) {
  .c8 {
    margin-left: 120px;
  }
}

<div
    className="c0"
  >
    <div
      className="MuiPaper-root MuiCard-root c1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiCardMedia-root c2"
        style={
          Object {
            "backgroundImage": "url(\\"http://www.publicdomainpictures.net/pictures/30000/nahled/maple-leaves-background.jpg\\")",
          }
        }
      />
      <div
        className=""
      >
        <div
          className="MuiCardContent-root "
        >
          <h2
            className="MuiTypography-root c3 MuiTypography-h6 MuiTypography-gutterBottom"
          >
            Malbin Chromerocket
          </h2>
          <p
            className="MuiTypography-root c3 MuiTypography-body1"
          >
            <span>
              Age: 
            </span>
            166
          </p>
          <p
            className="MuiTypography-root c3 MuiTypography-body1"
          >
            <span>
              Weight: 
            </span>
            35.88665
          </p>
          <p
            className="MuiTypography-root c3 MuiTypography-body1"
          >
            <span>
              Height: 
            </span>
            106.14395
          </p>
          <p
            className="MuiTypography-root c3 MuiTypography-body1"
          >
            <span>
              Hair Color: 
            </span>
            Red
          </p>
          <p
            className="MuiTypography-root c4 MuiTypography-body1"
          >
            Friends
          </p>
          <p
            className="MuiTypography-root c5 MuiTypography-body1"
          >
            Fizwood Voidtossle
          </p>
        </div>
        <div
          className="c6"
        >
          <div
            className="c7"
            onClick={[Function]}
          >
            <p
              className="MuiTypography-root c3 MuiTypography-body1"
            >
              <span>
                Profession
              </span>
            </p>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root c8"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
          <div
            className="c9"
          >
            <p
              className="MuiTypography-root c3 MuiTypography-body1"
            >
              Cook
            </p>
            <p
              className="MuiTypography-root c3 MuiTypography-body1"
            >
              Baker
            </p>
            <p
              className="MuiTypography-root c3 MuiTypography-body1"
            >
              Miner
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
